pkg/reconciler/apis/apiexportendpointslice/apiexportendpointslice_reconcile.go

```diff
@@ -64,35 +68,103 @@ func (r *endpointsReconciler) reconcile(ctx context.Context, apiExportEndpointSl
 	}
 	apiExport, err := r.getAPIExport(apiExportPath, apiExportEndpointSlice.Spec.APIExport.Name)
 	if err != nil {
+		reason := apisv1alpha1.InternalErrorReason
 		if errors.IsNotFound(err) {
 			// Don't keep the endpoints if the APIExport has been deleted
 			apiExportEndpointSlice.Status.APIExportEndpoints = nil
+			// No need to try again
+			err = nil
+			reason = apisv1alpha1.APIExportNotFoundReason
+		}
+		conditions.MarkFalse(
+			apiExportEndpointSlice,
+			apisv1alpha1.APIExportValid,
+			reason,
+			conditionsv1alpha1.ConditionSeverityError,
+			"Error getting APIExport %s|%s",
+			apiExportPath,
+			apiExportEndpointSlice.Spec.APIExport.Name,
+		)
+		conditions.MarkFalse(
+			apiExportEndpointSlice,
+			apisv1alpha1.APIExportEndpointSliceURLsReady,
```

Comment:
so a single transient error can flip this condition `false`, should we do better?

Reply:
If it is an error, which is recoverable the condition is indeed set to false because the reconciliation was not successful but it is queued again. If it is transient it flips back to true when the condition has gone.
The alternative is to set it as "unknown"

Reply:
set to unknown